api/src/main/java/org/openmrs/module/openhmis/inventory/InventoryModuleActivator.java

```diff
@@ -13,37 +13,188 @@
+		Item newItem = new Item();
+		try {
+			SimpleDateFormat sdf = new SimpleDateFormat(DATE_FORMAT);
+			if (StringUtils.isNotEmpty(lineValues[0])) {
+				newItem.setId(Integer.valueOf(lineValues[0]));
+			}
+			if (StringUtils.isEmpty(lineValues[1])) {
+				throw new InvalidInventoryDataException("Department value missing in input file");
+			}
+			newItem.setName(lineValues[1]);
+			newItem.setDescription(lineValues[2]);
+			newItem.setDepartment(getDepartment(lineValues[3]));
+			newItem.setDefaultPrice(getDefaultItemPrice(lineValues[4]));
+			newItem.setHasExpiration(getBooleanFlagFromString(lineValues[5]));
+			newItem.setConcept(getConcept(lineValues[6]));
+			newItem.setCreator(getUser(lineValues[7], false));
+			if (StringUtils.isEmpty(lineValues[8])) {
+				throw new InvalidInventoryDataException("Date Created value missing in input file");
+			}
+			newItem.setDateCreated(sdf.parse(lineValues[8]));
+			newItem.setChangedBy(getUser(lineValues[9], true));
+			newItem.setDateChanged(StringUtils.isEmpty(lineValues[10]) ? null : sdf.parse(lineValues[10]));
+			newItem.setRetired(getBooleanFlagFromString(lineValues[11]));
+			newItem.setRetiredBy(getUser(lineValues[12], true));
+			newItem.setDateRetired(StringUtils.isEmpty(lineValues[13]) ? null : sdf.parse(lineValues[13]));
+			newItem.setRetireReason(lineValues[14]);
+			if (StringUtils.isEmpty(lineValues[15])) {
+				throw new InvalidInventoryDataException("UUID value missing in input file");
+			}
+			newItem.setUuid(lineValues[15]);
+			newItem.setHasPhysicalInventory(getBooleanFlagFromString(lineValues[16]));
+			newItem.setDefaultExpirationPeriod(StringUtils.isEmpty(lineValues[17]) ? null : Integer.valueOf(lineValues[17]));
+			newItem.setConceptAccepted(getBooleanFlagFromString(lineValues[18]));
+			newItem.setMinimumQuantity(StringUtils.isEmpty(lineValues[19]) ? null : Integer.valueOf(lineValues[19]));
+			newItem.setBuyingPrice(BigDecimal.valueOf(Long.parseLong(lineValues[20])));
+
+		} catch (NumberFormatException ex) {
+			LOG.error("Invalid value found in input file", ex);
+			throw new InvalidInventoryDataException("Invalid value found in input file" + ex.getMessage());
+		} catch (ParseException e) {
+			LOG.error("Invalid Date format for Creation Date field", e);
+			throw new InvalidInventoryDataException("Invalid date format in input file" + e.getMessage());
+		}
+		return newItem;
+	}
+
+	private User getUser(String userId, boolean optional) {
+		if (StringUtils.isEmpty(userId) && optional) {
+			return null;
+		}
+		User creator = new User();
+		creator.setId(Integer.valueOf(userId));
+		return creator;
+	}
+
+	private Concept getConcept(String conceptId) {
+		if (StringUtils.isEmpty(conceptId)) {
+			return null;
+		}
+		Concept concept = new Concept();
+		concept.setConceptId(Integer.valueOf(conceptId));
+		return concept;
+	}
+
+	private boolean getBooleanFlagFromString(String lineValue) {
+		return StringUtils.isNotEmpty(lineValue) && "1".equalsIgnoreCase(lineValue);
```

Comment:
Why not use "true" and "false" strings? or "yes"/"no" (with ignore case)?

Reply:
Would be clearer if in CSV, the user enters Yes/No.. but when you save it in DB.. u use 0/1.